### DISCUSSION TITLE
Slight improvement regarding the git handles completion source

### DISCUSSION THREAD

**Discussion by Nelyah** [wincent/wincent#128] [Show and tell]
Hey @wincent, 

I watched your [video about adding a custom nvim-cmp source](https://youtu.be/gAsYolNrjtQ) and got inspired to look up your solution. It felt really clunky that I had to maintain a `.json` files with the relevant handles. Especially if you switch between different repos of different teams (or open-source repos). 

I committed this to my dotfiles: https://github.com/Nelyah/dotfiles/commit/f4cdb6a3f8a94e79b4348272f558140ef1fb5b1a (check out the `.config/nvim/lua/completion-sources/handles.lua` file from that commit). Using [plenary](https://github.com/nvim-lua/plenary.nvim) I'm gathering all users from the current repo I'm in, marshalling this into `json` format and then feeding the output to the function you made. 

All the work happens asynchronously. I feel that since a lot of the work originally comes from you, you could be interested in that solution as well! Thanks for showcasing it in the first place 😁

**Comment by wincent** [comment]
This is great @Nelyah; I think it proves exactly the point that I was trying to make in my video — namely, that with a bit of Lua, you can put together an autocomplete that pretty much fits your specific usecase (whatever that may be) like a glove.

**Reply by Nelyah** [reply]
Yes, I definitely agree! Thanks a lot for taking the time to make those videos, they are always very informative and I'm always learning a trick or two.
